mlflow/tensorflow/__init__.py

```diff
@@ -1257,6 +1265,29 @@ def __restore_generator(prev_generator):
                 if log_models:
                     _log_keras_model(history, args)
 
+                if log_datasets:
+                    try:
+                        # create a CodeDatasetSource
+                        context_tags = context_registry.resolve_tags()
+                        source = CodeDatasetSource(
+                            mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                            mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                        )
+                        training_data = kwargs["x"] if "x" in kwargs else args[0]
+                        validation_data = (
+                            kwargs["validation_data"] if "validation_data" in kwargs else None
+                        )
```

Comment:
Couldn't `validation_data` also be `args[7]`? https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit